adding tasks with tests

pdziugas opened the pull request:
Added solved tasks with their tests.

pdziugas committed:
adding tasks with tests

visma-zanas-stundys commented on .gitignore:
```diff
@@ -0,0 +1 @@
+node_modules
```
Consider adding coverage reports as well :)

visma-zanas-stundys commented on tasks/1.js:
```diff
@@ -0,0 +1,26 @@
+
+  return list;
+}
+
+console.log(rangeES5(4, 10)); // returns [ 4, 5, 6, 7, 8, 9 ]
+
```
You should keep debugging statements out of commited code ;)

visma-zanas-stundys commented on tasks/4.test.js:
```diff
@@ -0,0 +1,13 @@
+
+describe("Task #4", () => {
+  describe("removeDuplicates()", () => {
+    it("should remove duplicates in an array", () => {
+      const array = ["one", "two", "three", "one", "four", "five", "six"];
+      const result = removeDuplicates(array);
+      const expected = ["one", "two", "three", "four", "five", "six"];
+
+      expect(result).toStrictEqual(expected);
+    });
```
Try `[10, 10, 10]`

visma-zanas-stundys commented on tasks/5.test.js:
```diff
@@ -0,0 +1,12 @@
+describe("Task #5", () => {
+  describe("wordSearch()", () => {
+    it("should return true for searching case insensitive word", () => {
+      const result = wordSearch();
```
No arguments passed. Also please note that it should be full-word search ;) Meaning: `wordSearch('foobar', 'foo')` should return false

LukasTyla replied:
Yeah, try the following test cases:
```js
wordSearch('foo', 'foobar') // false
wordSearch('foo', 'Foo bar') // true
wordSearch('foo', ' foo bar') // true
```
Consider looking into word boundary. 😉

visma-zanas-stundys commented on tasks/7.test.js:
```diff
@@ -0,0 +1,12 @@
+      const result = sevenAte9("797");
+      const expected = "77";
+
+      expect(result).toBe(expected);
```
Consider using other types of matchers, for example:

```
expect(result).not.stringContaining('797');
```

visma-zanas-stundys commented on tasks/9.js:
```diff
@@ -0,0 +1,10 @@
+let arr = [10, 6, [4, 8], 3, [6, 5, [9]]];
+// Infinity flattens array as many times as needed
+const sumNestedArray = () =>
+  arr.flat(Infinity).reduce((accumulator, currentValue) => accumulator + currentValue);
```
Functions that rely on variables outside of scope might be difficult to test. Consider making functions to accept arguments, so that various cases could be tested

LukasTyla commented on tasks/10.js:
```diff
@@ -0,0 +1,70 @@
+var CalculatorES5 = function () {
```
Nitpick: nor this, nor ES6 variant accept a variable, but you are initialising these objects by also passing a `0` as an argument. 😃

LukasTyla commented on tasks/2.js:
```diff
@@ -0,0 +1,24 @@
+// ES5
+var listOfNumbersES5 = [1, 2, 3, 4, 5, 6];
+
+function sumES5() {
```
Nitpick, we expect this function to accept the array itself and operate on it instead of using hardcoded values. 😉

LukasTyla commented on tasks/4.test.js:
```diff
@@ -0,0 +1,13 @@
+  describe("removeDuplicates()", () => {
+    it("should remove duplicates in an array", () => {
+      const array = ["one", "two", "three", "one", "four", "five", "six"];
+      const result = removeDuplicates(array);
```
Your function does not accept any arguments, it operates on it's own constant of array.

LukasTyla commented on tasks/5.test.js:
```diff
@@ -0,0 +1,12 @@
+const { wordSearch } = require("./5");
+
+describe("Task #5", () => {
+  describe("wordSearch()", () => {
```
Wrapping the test suite in an additional describe for `task` is sort of redundant as you have no other functions to test. 🤔

pdziugas committed:
PR changes:
- added coverage reports to gitignore
- removed debugging statements out of the code
- word search now works as intended (searches for full case insensitive words)

pdziugas committed:
PR change:
- now actually using passed array in a sumNestedArray function
- added another unit test for sumNestedArray
